addon/-private/execution_context/native-dom-context.js

```diff
@@ -0,0 +1,138 @@
+import $ from '-jquery';
+
+import {
+  click,
+  triggerEvent,
+  keyEvent
+} from 'ember-native-dom-helpers';
+
+import {
+  guardMultiple,
+  buildSelector,
+  findClosestValue
+} from '../helpers';
+import {
+  fillElement
+} from './helpers';
+import {
+  ELEMENT_NOT_FOUND,
+  throwBetterError
+} from '../better-errors';
+
+const KEYBOARD_EVENT_TYPES = ['keydown', 'keypress', 'keyup'];
+
+export default function ExecutionContext(pageObjectNode, testContext) {
+  this.pageObjectNode = pageObjectNode;
+  this.testContext = testContext;
+}
+
+ExecutionContext.prototype = {
+  run(cb) {
+    return cb(this);
+  },
+
+  runAsync() {
+    throw new Error('not implemented');
+  },
+
+  click(selector, container) {
+    const el = this.$(selector, container)[0];
+    click(el);
+  },
+
+  fillIn(selector, container, options, content) {
+    let elements = this.$(selector, container).toArray();
+
+    elements.forEach((el) => {
+      fillElement(el, content, {
```

Comment:
Todo: 
1. Update function signature of `fillElement` to indicate that the first param is an element
2. Update the usage of `fillElement` in the other places where it is used so that we are always passing an element and not a jQuery selection.